design/work_item_link_type.go

```diff
@@ -52,6 +52,8 @@ See also see http://jsonapi.org/format/#document-resource-object-attributes`)
 	a.Attribute("version", d.Integer, "Version for optimistic concurrency control (optional during creating)", func() {
```

Comment:
Update and Delete would have the same Conditional. But behave a little bit differently maybe... 

I guess it's more normal to use If-Match or If-Unmodified-Since to do state-changing request like PUT/PATCH/POST.

Don't perform the Update unless if matches, etc..

Happy to get that bit in a separate PR :)